R/plot_sankey.R

```diff
@@ -0,0 +1,99 @@
+#' Create a sankey plot
+#'
+#' @param data A data frame like [toy_data].
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' plot_sankey(toy_data)
+plot_sankey <- function(data) {
+  data_links <- data |>
+    mutate(
+      source = "bank",
+      target = "PCTR_risk_category",
+      value = "amount",
+      middle_node2 = "tilt_sec"
+    )
+
+  links <- data_links |>
+    select(
+      source = "bank",
+      target = "tilt_sec",
+      value = "amount",
+      group = "PCTR_risk_category"
+    )
+  # FIXME: This is always TRUE because the first code chunks adds the column
+  # `middle_node2`. That is, the `else` branch never runs.
+  # TODO: Replace with hasName()
+  if ("middle_node2" %in% names(data_links)) {
+    links <- data_links |>
+      select(
+        "bank",
+        source = "bank",
+        target = "middle_node2",
+        value = "amount",
+        group = "PCTR_risk_category"
+      )
```

Comment:
FIXME: Is this dead code? The object `links` is defined below again so this one isn't used.